client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
 
 	render() {
 		const { suggestion, translate } = this.props,
-			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
+			domain = suggestion.domain_name,
+			isAdded = hasDomainInCart( this.props.cart, domain ),
 			domainFlags = [];
+
 		let buttonClasses, buttonContent;
 
-		if ( suggestion.domain_name ) {
+		if ( domain ) {
 			const newTLDs = [];
+			const a8cTLDs = [ '.ca', '.de', '.fr' ];
+			const tld = domain.substring( domain.indexOf( '.' ) );
```

Comment:
I have mixed feelings about this - while it does nicely fix the issue with the previous approach (not recognizing second level TLDs), it won't work with subdomains... but that should not be an issue, since showing a subdomain here would be pointless. Unless in the future we'll show the subdomain here to suggest it for mapping - but at that point I'm sure we'll need to refactor some other code ;)
Sooo, it's good, but I'm not happy about it 😅 (maybe because I'm jealous that I did not come up with this, but instead the more complicated one 😉 )